jdaviz/configs/default/plugins/export/export.py

```diff
@@ -258,6 +259,42 @@ def _is_filename_changed(self, event):
         # Clear overwrite warning when user changes filename
         self.overwrite_warn = False
 
+    def _update_subset_format_disabled(self):
+        new_items = []
+        if self.subset.selected is not None:
+            subset = self.app.get_subsets(self.subset.selected)
+            if self.app._is_subset_spectral(subset[0]):
+                good_formats = ["ecsv"]
+            else:
+                good_formats = ["fits", "reg"]
+            for item in self.subset_format_items:
+                if item["label"] in good_formats:
+                    item["disabled"] = False
+                else:
+                    item["disabled"] = True
+                    if item["label"] == self.subset_format.selected:
+                        self.subset_format.selected = good_formats[0]
+                new_items.append(item)
+        self.subset_format_items = []
+        self.subset_format_items = new_items
+
+    @observe('subset_format_selected')
+    def _disable_subset_format_combo(self, event):
+        # Disable selecting a bad subset+format combination from the API
+        if self.subset.selected == '' or self.subset.selected is None:
+            return
+        subset = self.app.get_subsets(self.subset.selected)
+        bad_combo = False
+        if self.app._is_subset_spectral(subset[0]):
+            if event['new'] != "ecsv":
+                bad_combo = True
+        elif event['new'] == "ecsv":
+            bad_combo = True
+
+        if bad_combo:
+            raise ValueError(f"Cannot export {self.subset.selected} in "
+                             f"{self.subset_format.selected.upper()} format")

```

Comment:
This doesn't revert the actual value, and so leaves the plugin in a broken state at which point clicking export or calling `export()` gives an uninformative traceback.  I suspect the code to revert the selection would be messy and perhaps confusing behavior, but maybe we can keep this error message, still show the red text and disable the export button, and then raise that red text as an error on export (all that infrastructure should already exist for the other cases already 🤞 )

To reproduce:
* create a spectral subset and select in the UI
* using the API, request 'fits' as the format.  This error is raised but the value is still set.
* (subsequent attempts to set would not raise an error since the value hasn't changed)
* click or call export